### DISCUSSION TITLE
Bad UX and security risk when adding users to teams inside an organization

### DISCUSSION THREAD

**Discussion by Anwardo** [community/community#73029] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

When adding a user to a team inside of an organization it should list all the users inside your organization by default so you can easily select users who are actually inside your organization. Currently it forces you to search for users inside an autocomplete which won't even list users inside your organization first.

I feel like it probably shouldn't even list users outside or your organization by default but only using a toggle or having two separate autocompletes one for users inside your organization or and one for users outside of your organization. Accidentally adding users who are not in your organization may be a security risk.

Either way it's just a bad UX overall currently since it forces me to first check the user list of my organization to make sure I'm searching for the right user. Would be great if this could be fixed!

**Comment by oussamaef06** [comment]
Your suggestion for improving the user management interface in GitHub organizations is valid. It's essential for organization administrators to have a clear and user-friendly interface for managing team members, which includes selecting users within their organization. While I can't directly implement this change, I can offer some insights into how you can provide feedback to GitHub and suggest this improvement:

GitHub Community Forum: GitHub maintains a community forum where users can discuss topics, ask questions, and provide feedback. You can post your suggestion in the GitHub Community under the "GitHub Actions" or "GitHub Help" categories, or search for similar suggestions to upvote.

GitHub Support: You can also reach out to GitHub Support directly with your suggestion. They can help you with the submission process and provide guidance on how to suggest the feature or improvement.

GitHub Product Feedback: GitHub provides a "[GitHub Feedback](https://support.github.com/contact/feedback)" platform where you can submit product feedback and feature requests. You can access it by visiting the following link: GitHub Feedback.

Feature Requests on GitHub: Some users choose to create feature request repositories on GitHub where they document and discuss feature requests. Others can participate in the discussion and show support for the request.

Participate in GitHub's Early Access Programs: GitHub occasionally runs early access programs for new features. By participating in these programs, you can provide direct feedback on new features or improvements.

When providing your feedback, be sure to include a clear description of the problem you're encountering and how your proposed improvement would address it. The more detailed and compelling your feedback, the more likely it is to receive attention from GitHub's product and development teams.

While I can't guarantee that your suggestion will be implemented, GitHub does take user feedback seriously and considers it in their ongoing development efforts. Your input can contribute to making GitHub's user interface more user-friendly and secure.

**Comment by MaxDNG** [comment]
Yes please! Having to type the full user name when adding a user to a Team or Repo is annoying, and I'm always worried I will add someone from outside the organization by mistake.

**Comment by sharprs83** [comment]
Even when our org doesn't allow external users to be added, it's still strange that external users are listed in the results—they're dimmed but visible.

Plus, it'd be super helpful if we could search by _name_. As if I know the bizarre usernames some people choose...
